service/tbc/rpc.go

```diff
@@ -545,7 +560,43 @@ func (s *Server) handleTxBroadcastRawRequest(ctx context.Context, req *tbcapi.Tx
 		return &tbcapi.TxBroadcastResponse{Error: e.ProtocolError()}, e
 	}
 
-	return &tbcapi.TxBroadcastResponse{TxID: txid}, nil
+	return &tbcapi.TxBroadcastRawResponse{TxID: txid}, nil
+}
+
+func (s *Server) handleBlockInsertRequest(ctx context.Context, req *tbcapi.BlockInsertRequest) (any, error) {
+	log.Tracef("handleBlockInsertRequest")
+	defer log.Tracef("handleBlockInsertRequest exit")
+
+	_, err := s.db.BlockInsert(ctx, btcutil.NewBlock(req.Block))
```

Comment:
As `req.Block` is a pointer, it is possible that the JSON payload could be `{"block": null}`.
Adding a nil-check above this would prevent possible panics; however I am not sure how `btcutil.NewBlock` will handle `nil`.

Reply:
Thanks!